Core: Add JSONMessagePart for Hint Status (Hint Priority) (#4387)
ArchipelagoMW/Archipelago: docs/network protocol.md

    color: Optional[str] # only available if type is a color
    flags: Optional[int] # only available if type is an item_id or item_name
    player: Optional[int] # only available if type is either item or location

>     hint_status: Optional[int] # only available if type is hint_status

Reviewer: I guess this should be Optional[HintStatus] rather than int

Reviewer 2: ```suggestion
    hint_status: Optional[[HintStatus](#HintStatus)] # only available if type is hint_status
```
aye, that would seem more correct.

>     hint_status: Optional[[HintStatus](#HintStatus)] # only available if type is hint_status

Reviewer 2: ```suggestion
    hint_status: Optional[HintStatus] # only available if type is hint_status
```

| location_id | Location ID, should be resolved to Location Name |
| location_name | Location Name, not currently used over network, but supported by reference Clients. |
| entrance_name | Entrance Name. No ID mapping exists. |

> | hint_status | The status of the hint (i.e. found, priority). Both `text` and `hint_status` are given. |

Reviewer 2: ```suggestion
| hint_status | The [HintStatus](#HintStatus) of the hint. Both `text` and `hint_status` are given. |
```

> | hint_status | The status of the hint (i.e. found, priority). See [HintStatus](#HintStatus) for details. Both `text` and `hint_status` are given. |

Reviewer 2: ```suggestion
| hint_status | The status of the hint (i.e. found, priority). Both `text` and `hint_status` are given. |
```
Don't need this bit here, if you apply my other suggestion as well.
